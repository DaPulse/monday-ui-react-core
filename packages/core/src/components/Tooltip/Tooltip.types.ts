export type TooltipPositions = "top" | "right" | "bottom" | "left";

export type TooltipTheme = "dark" | "primary";

export type MoveBy = {
  main?: number;
  secondary?: number;
};
