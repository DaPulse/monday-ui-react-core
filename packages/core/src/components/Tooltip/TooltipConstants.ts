export enum TooltipArrowPosition {
  BEGIN = "begin",
  CENTER = "center",
  END = "end"
}

export type TooltipTheme = "dark" | "primary";
