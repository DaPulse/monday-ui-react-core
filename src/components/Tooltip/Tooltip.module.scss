@import "../../styles/themes";
@import "../../styles/typography";
@import "../../styles/global-css-settings";

$tooltip-z-index: 9999999999999999;

.tooltip {
  position: relative;
  display: inline-block;
  border-radius: var(--tooltip-border-radius, var(--border-radius-small));
  padding: var(--tooltip-padding, #{var(--spacing-small) var(--spacing-medium)});
  @include theme-prop(box-shadow, box-shadow-medium);
  @include font-input();
  max-width: var(--tooltip-max-width, 50vw);
  word-break: break-word;
}

.tooltip.paddingSizeMd {
  border-radius: var(--border-radius-medium);
  padding: var(--spacing-medium);
  @include font-input();
}

.tooltip a.tooltipWhiteLink {
  color: #fff;
}

.dark,
.arrow.dark {
  @include theme-prop(background-color, inverted-color-background);
  @include theme-prop(color, text-color-on-inverted);
}

.white,
.arrow.white {
  background-color: var(--color-snow_white);
  color: var(--color-blackish);
  @include theme-prop(box-shadow, box-shadow-medium);
}

.primary,
.arrow.primary {
  @include theme-prop(background-color, primary-color);
  @include theme-prop(color, text-color-on-primary);
}

.success,
.arrow.success {
  @include theme-prop(background-color, positive-color);
  @include theme-prop(color, text-color-on-primary);
}

.surface,
.arrow.surface {
  background-color: #5b607a;
  color: var(--color-snow_white);
}

.private,
.arrow.private {
  @include theme-prop(background-color, private-color);
  @include theme-prop(color, text-color-on-primary);
}

.share,
.arrow.share {
  @include theme-prop(background-color, shareable-color);
  @include theme-prop(color, text-color-on-primary);
}

.error,
.arrow.error {
  @include theme-prop(background-color, negative-color);
  @include theme-prop(color, text-color-on-primary);
}
