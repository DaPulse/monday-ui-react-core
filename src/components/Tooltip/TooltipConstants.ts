export enum TooltipArrowPosition {
  BEGIN = "begin",
  CENTER = "center",
  END = "end"
}

export enum TooltipTheme {
  DARK = "dark",
  SUCCESS = "success",
  ERROR = "error",
  SHARE = "share",
  PRIVATE = "private",
  SURFACE = "surface",
  PRIMARY = "primary"
}
