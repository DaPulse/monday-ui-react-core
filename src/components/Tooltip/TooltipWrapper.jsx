import React from "react";
import PropTypes from "prop-types";
import Tooltip from "./Tooltip";

const TooltipWrapper = ({ children, wrapWithTooltip, tooltipProps }) => {
  if (!wrapWithTooltip) {
    return children;
  }

  return <Tooltip {...tooltipProps}>{children}</Tooltip>;
};

TooltipWrapper.propTypes = {
  children: PropTypes.element,
  wrapWithTooltip: PropTypes.bool,
  tooltipProps: PropTypes.shape(Tooltip.propTypes)
};
TooltipWrapper.defaultProps = {
  children: undefined,
  wrapWithTooltip: true,
  tooltipProps: {}
};

export default TooltipWrapper;
