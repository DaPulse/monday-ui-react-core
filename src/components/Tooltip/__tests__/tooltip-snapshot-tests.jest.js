import React from "react";
import renderer from "react-test-renderer";
import Tooltip from "../Tooltip";

describe("Tooltip renders correctly", () => {
  it("with empty props", () => {
    const tree = renderer.create(<Tooltip shouldShowOnMount />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with end arrowPosition", () => {
    const tree = renderer.create(<Tooltip shouldShowOnMount arrowPosition="end" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with theme", () => {
    const tree = renderer.create(<Tooltip shouldShowOnMount theme={Tooltip.themes.Error} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with position", () => {
    const tree = renderer.create(<Tooltip shouldShowOnMount position={Tooltip.positions.LEFT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with justify", () => {
    const tree = renderer.create(<Tooltip shouldShowOnMount justify={Tooltip.justifyTypes.END} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with withoutDialog", () => {
    const tree = renderer.create(<Tooltip withoutDialog />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("without arrow", () => {
    const tree = renderer.create(<Tooltip tip={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with y", () => {
    const tree = renderer.create(<Tooltip />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with y", () => {
    const tree = renderer.create(<Tooltip />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});