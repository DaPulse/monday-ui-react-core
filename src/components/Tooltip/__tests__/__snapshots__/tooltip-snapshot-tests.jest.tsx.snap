// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders correctly with end arrowPosition 1`] = `
Array [
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />,
  <span
    className="contentWrapper monday-style-dialog-content-wrapper"
    data-popper-reference-hidden={false}
    onClickCapture={[Function]}
    style={
      Object {
        "left": "0",
        "position": "absolute",
        "top": "0",
      }
    }
  >
    <div
      className="contentComponent monday-style-dialog-content-component top top edgeCenter edge-center hasTooltip has-tooltip"
    >
      <div
        className="tooltip dark paddingSizeUndefined"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        test
      </div>
      <div
        className="arrow monday-style-tooltip-arrow arrow dark paddingSizeUndefined"
        data-placement="top"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "position": "absolute",
          }
        }
      />
    </div>
  </span>,
]
`;

exports[`Tooltip renders correctly with hideWhenReferenceHidden 1`] = `
Array [
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />,
  <span
    className="contentWrapper monday-style-dialog-content-wrapper"
    data-popper-reference-hidden={null}
    onClickCapture={[Function]}
    style={
      Object {
        "left": "0",
        "position": "absolute",
        "top": "0",
      }
    }
  >
    <div
      className="contentComponent monday-style-dialog-content-component top top edgeCenter edge-center hasTooltip has-tooltip"
    >
      <div
        className="tooltip dark paddingSizeUndefined"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        test
      </div>
      <div
        className="arrow monday-style-tooltip-arrow arrow dark paddingSizeUndefined"
        data-placement="top"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "position": "absolute",
          }
        }
      />
    </div>
  </span>,
]
`;

exports[`Tooltip renders correctly with justify 1`] = `
Array [
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />,
  <span
    className="contentWrapper monday-style-dialog-content-wrapper"
    data-popper-reference-hidden={false}
    onClickCapture={[Function]}
    style={
      Object {
        "left": "0",
        "position": "absolute",
        "top": "0",
      }
    }
  >
    <div
      className="contentComponent monday-style-dialog-content-component top top edgeEnd edge-end hasTooltip has-tooltip"
    >
      <div
        className="tooltip dark paddingSizeUndefined"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        test
      </div>
      <div
        className="arrow monday-style-tooltip-arrow arrow dark paddingSizeUndefined"
        data-placement="top"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "position": "absolute",
          }
        }
      />
    </div>
  </span>,
]
`;

exports[`Tooltip renders correctly with position 1`] = `
Array [
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />,
  <span
    className="contentWrapper monday-style-dialog-content-wrapper"
    data-popper-reference-hidden={false}
    onClickCapture={[Function]}
    style={
      Object {
        "left": "0",
        "position": "absolute",
        "top": "0",
      }
    }
  >
    <div
      className="contentComponent monday-style-dialog-content-component left left edgeCenter edge-center hasTooltip has-tooltip"
    >
      <div
        className="tooltip dark paddingSizeUndefined"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        test
      </div>
      <div
        className="arrow monday-style-tooltip-arrow arrow dark paddingSizeUndefined"
        data-placement="left"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "position": "absolute",
          }
        }
      />
    </div>
  </span>,
]
`;

exports[`Tooltip renders correctly with style 1`] = `
Array [
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />,
  <span
    className="contentWrapper monday-style-dialog-content-wrapper"
    data-popper-reference-hidden={false}
    onClickCapture={[Function]}
    style={
      Object {
        "left": "0",
        "position": "absolute",
        "top": "0",
      }
    }
  >
    <div
      className="contentComponent monday-style-dialog-content-component top top edgeCenter edge-center hasTooltip has-tooltip"
    >
      <div
        className="tooltip dark paddingSizeUndefined"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "width": "200px",
          }
        }
      >
        test
      </div>
      <div
        className="arrow monday-style-tooltip-arrow arrow dark paddingSizeUndefined"
        data-placement="top"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "position": "absolute",
          }
        }
      />
    </div>
  </span>,
]
`;

exports[`Tooltip renders correctly with theme 1`] = `
Array [
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />,
  <span
    className="contentWrapper monday-style-dialog-content-wrapper"
    data-popper-reference-hidden={false}
    onClickCapture={[Function]}
    style={
      Object {
        "left": "0",
        "position": "absolute",
        "top": "0",
      }
    }
  >
    <div
      className="contentComponent monday-style-dialog-content-component top top edgeCenter edge-center hasTooltip has-tooltip"
    >
      <div
        className="tooltip error paddingSizeUndefined"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        test
      </div>
      <div
        className="arrow monday-style-tooltip-arrow arrow error paddingSizeUndefined"
        data-placement="top"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "position": "absolute",
          }
        }
      />
    </div>
  </span>,
]
`;

exports[`Tooltip renders correctly with withoutDialog 1`] = `
<div
  className="tooltip dark paddingSizeUndefined"
>
  test
</div>
`;

exports[`Tooltip renders correctly without arrow 1`] = `
Array [
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />,
  <span
    className="contentWrapper monday-style-dialog-content-wrapper"
    data-popper-reference-hidden={false}
    onClickCapture={[Function]}
    style={
      Object {
        "left": "0",
        "position": "absolute",
        "top": "0",
      }
    }
  >
    <div
      className="contentComponent monday-style-dialog-content-component top top edgeCenter edge-center"
    >
      <div
        className="tooltip dark paddingSizeUndefined"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        test
      </div>
    </div>
  </span>,
]
`;
